From 896c04b6cc2ef73573cfcc4b29d72942c9aaa4c0 Mon Sep 17 00:00:00 2001
From: Martin Daser <martin.daser@intel.com>
Date: Mon, 14 May 2018 15:49:31 +0200
Subject: [PATCH] default configuration: pine64_plus board

this is a default configuration for the Pine64+ board.
it is not a minimal configuration though, but includes
- systemd
- glibc
- avahi
- et al

Signed-off-by: Martin Daser <martin.daser@web.de>
---
 board/pine64/pine64_plus/boot.cmd     |  6 ++++
 board/pine64/pine64_plus/genimage.cfg | 39 ++++++++++++++++++++++++
 board/pine64/pine64_plus/readme.txt   | 38 +++++++++++++++++++++++
 configs/pine64_plus_defconfig         | 57 +++++++++++++++++++++++++++++++++++
 4 files changed, 140 insertions(+)
 create mode 100644 board/pine64/pine64_plus/boot.cmd
 create mode 100644 board/pine64/pine64_plus/genimage.cfg
 create mode 100644 board/pine64/pine64_plus/readme.txt
 create mode 100644 configs/pine64_plus_defconfig

diff --git a/board/pine64/pine64_plus/boot.cmd b/board/pine64/pine64_plus/boot.cmd
new file mode 100644
index 0000000..79ac4d0
--- /dev/null
+++ b/board/pine64/pine64_plus/boot.cmd
@@ -0,0 +1,6 @@
+setenv bootargs console=ttyS0,115200 earlyprintk root=/dev/mmcblk0p2 rootwait
+
+fatload mmc 0 $kernel_addr_r Image
+fatload mmc 0 $fdt_addr_r sun50i-a64-pine64-plus.dtb
+
+booti $kernel_addr_r - $fdt_addr_r
diff --git a/board/pine64/pine64_plus/genimage.cfg b/board/pine64/pine64_plus/genimage.cfg
new file mode 100644
index 0000000..60dac9d
--- /dev/null
+++ b/board/pine64/pine64_plus/genimage.cfg
@@ -0,0 +1,39 @@
+image boot.vfat {
+	vfat {
+		files = {
+			"Image",
+			"sun50i-a64-pine64-plus.dtb",
+			"boot.scr"
+		}
+	}
+	size = 64M
+}
+
+image sdcard.img {
+	hdimage {
+	}
+
+	partition spl {
+		in-partition-table = "no"
+		image = "sunxi-spl.bin"
+		offset = 8192
+	}
+
+	partition u-boot {
+		in-partition-table = "no"
+		image = "u-boot.itb"
+		offset = 40K
+		size = 1M # 1MB - 40K
+	}
+
+	partition boot {
+		partition-type = 0xC
+		bootable = "true"
+		image = "boot.vfat"
+	}
+
+	partition rootfs {
+		partition-type = 0x83
+		image = "rootfs.ext4"
+	}
+}
diff --git a/board/pine64/pine64_plus/readme.txt b/board/pine64/pine64_plus/readme.txt
new file mode 100644
index 0000000..85226fd
--- /dev/null
+++ b/board/pine64/pine64_plus/readme.txt
@@ -0,0 +1,38 @@
+Intro
+=====
+
+This default configuration will allow you to start experimenting with the
+buildroot environment for the Pine64. With the current configuration
+it will bring-up the board, and allow access through the serial console.
+
+Pine64 link:
+https://www.pine64.org/?page_id=1194
+
+Wiki link:
+https://openedev.amarulasolutions.com/display/ODWIKI/Pine64
+
+This configuration uses U-Boot mainline and kernel mainline.
+
+How to build
+============
+
+    $ make pine64_defconfig
+
+    $ make
+
+Note: you will need access to the internet to download the required
+sources.
+
+How to write the SD card
+========================
+
+Once the build process is finished you will have an image called "sdcard.img"
+in the output/images/ directory.
+
+Copy the bootable "sdcard.img" onto an SD card with "dd":
+
+  $ sudo dd if=output/images/sdcard.img of=/dev/sdX
+  $ sudo sync
+
+Insert the micro SDcard in your Pine64 and power it up. The console
+is on the serial line, 115200 8N1.
diff --git a/configs/pine64_plus_defconfig b/configs/pine64_plus_defconfig
new file mode 100644
index 0000000..08e085d
--- /dev/null
+++ b/configs/pine64_plus_defconfig
@@ -0,0 +1,57 @@
+BR2_aarch64=y
+BR2_ARM_FPU_VFPV4=y
+BR2_TOOLCHAIN_BUILDROOT_GLIBC=y
+BR2_TOOLCHAIN_BUILDROOT_CXX=y
+BR2_TARGET_GENERIC_HOSTNAME="pine64"
+BR2_TARGET_GENERIC_ISSUE="Welcome to PINE64"
+BR2_INIT_SYSTEMD=y
+BR2_TARGET_GENERIC_ROOT_PASSWD="root"
+BR2_SYSTEM_DHCP="eth0"
+BR2_ENABLE_LOCALE_WHITELIST="C en_US de_DE"
+BR2_GENERATE_LOCALE="C en_US de_DE"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="support/scripts/genimage.sh"
+BR2_ROOTFS_POST_SCRIPT_ARGS="-c board/pine64/pine64_plus/genimage.cfg"
+BR2_LINUX_KERNEL=y
+BR2_LINUX_KERNEL_CUSTOM_VERSION=y
+BR2_LINUX_KERNEL_CUSTOM_VERSION_VALUE="4.16.4"
+BR2_LINUX_KERNEL_USE_ARCH_DEFAULT_CONFIG=y
+BR2_LINUX_KERNEL_DTS_SUPPORT=y
+BR2_LINUX_KERNEL_INTREE_DTS_NAME="allwinner/sun50i-a64-pine64-plus"
+BR2_LINUX_KERNEL_EXT_FBTFT=y
+BR2_PACKAGE_LINUX_TOOLS_GPIO=y
+BR2_PACKAGE_SDL=y
+BR2_PACKAGE_LINUXCONSOLETOOLS=y
+BR2_PACKAGE_AVAHI=y
+BR2_PACKAGE_AVAHI_DAEMON=y
+BR2_PACKAGE_AVAHI_LIBDNSSD_COMPATIBILITY=y
+BR2_PACKAGE_IPUTILS=y
+BR2_PACKAGE_MINIDLNA=y
+BR2_PACKAGE_OPENSSH=y
+BR2_PACKAGE_USHARE=y
+BR2_PACKAGE_TMUX=y
+BR2_TARGET_ROOTFS_EXT2=y
+BR2_TARGET_ROOTFS_EXT2_4=y
+BR2_TARGET_ROOTFS_EXT2_SIZE="700M"
+BR2_TARGET_ARM_TRUSTED_FIRMWARE=y
+BR2_TARGET_ARM_TRUSTED_FIRMWARE_CUSTOM_GIT=y
+BR2_TARGET_ARM_TRUSTED_FIRMWARE_CUSTOM_REPO_URL="https://github.com/apritzel/arm-trusted-firmware.git"
+BR2_TARGET_ARM_TRUSTED_FIRMWARE_CUSTOM_REPO_VERSION="aa75c8da415158a94b82a430b2b40000778e851f"
+BR2_TARGET_ARM_TRUSTED_FIRMWARE_PLATFORM="sun50iw1p1"
+BR2_TARGET_UBOOT=y
+BR2_TARGET_UBOOT_BUILD_SYSTEM_KCONFIG=y
+BR2_TARGET_UBOOT_CUSTOM_VERSION=y
+BR2_TARGET_UBOOT_CUSTOM_VERSION_VALUE="2017.11"
+BR2_TARGET_UBOOT_BOARD_DEFCONFIG="pine64_plus"
+BR2_TARGET_UBOOT_NEEDS_DTC=y
+BR2_TARGET_UBOOT_NEEDS_ATF_BL31=y
+BR2_TARGET_UBOOT_FORMAT_CUSTOM=y
+BR2_TARGET_UBOOT_FORMAT_CUSTOM_NAME="u-boot.itb"
+BR2_TARGET_UBOOT_SPL=y
+BR2_TARGET_UBOOT_SPL_NAME="spl/sunxi-spl.bin"
+BR2_TARGET_UBOOT_BOOT_SCRIPT=y
+BR2_TARGET_UBOOT_BOOT_SCRIPT_SOURCE="board/pine64/pine64_plus/boot.cmd"
+BR2_PACKAGE_HOST_DOSFSTOOLS=y
+BR2_PACKAGE_HOST_GENIMAGE=y
+BR2_PACKAGE_HOST_MTOOLS=y
+BR2_PACKAGE_HOST_UBOOT_TOOLS=y
+BR2_PACKAGE_HOST_UBOOT_TOOLS_FIT_SUPPORT=y
-- 
2.7.4

